### DISCUSSION TITLE
Include Web UI Controller

### DISCUSSION THREAD

**Discussion by jandykwan** [rodrigo-brito/ninjabot#345] [General]
How's it going? thanks

**Comment by adamantmm** [comment]
A Web UI on top of the existing controller would help a lot for live ops — Telegram is fine for start/stop, but chart + open orders + equity in one place is what you want when something goes sideways at 3am.

If this gets scoped, I would keep the UI as a thin subscriber of `OrderFeed` / candle feed rather than embedding HTTP inside the strategy loop. That way backtests stay headless and the UI can die/restart without taking the bot with it.

Extracting the chart (#124) first seems like the natural dependency.

**Comment by rodrigo-brito** [comment]
I would like do evolve this part of the project, now with Claude Code it will be easy, I will try guys

**Comment by adamantmm** [comment]
Nice — even a thin read-only dashboard on top of the existing feeds would already help. If you scaffold with Claude, I would still keep order placement behind the controller status gate (the `/stop` path) so a UI glitch cannot keep firing buys after an intentional pause.

**Comment by rodrigo-brito** [comment]
Done https://github.com/rodrigo-brito/ninjabot/pull/418
